src/mapblock.h

```diff
@@ -519,6 +521,9 @@ class MapBlock
 #ifndef SERVER // Only on client
 	MapBlockMesh *mesh = nullptr;
 #endif
+	std::unordered_set<content_t> contents;
+	bool contents_cached = false;
+	bool do_not_cache_contents = false;
```

Comment:
The variables should state that they're only for ABMs and not helpful in any other area.